pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -596,20 +596,20 @@ class Timestamp(_Timestamp):
             return Timestamp(datetime(year, month, day, hour or 0,
                                       minute or 0, second or 0,
                                       microsecond or 0, tzinfo),
-                             tz=tz)
+                             nanosecond=nanosecond, tz=tz)
         elif is_integer_object(freq):
             # User passed positional arguments:
             # Timestamp(year, month, day[, hour[, minute[, second[,
             # microsecond[, tzinfo]]]]])
             return Timestamp(datetime(ts_input, freq, tz, unit or 0,
                                       year or 0, month or 0, day or 0,
-                                      hour), tz=hour)
+                                      hour), nanosecond=minute, tz=hour)
```

Comment:
this doesn't seem right

Reply:
I tested over these two cases in `test_constructor_nanosecond` that would hit this path:

```
Timestamp(2000, 1, 1, 0, 0, 0, 0, None, 1)
Timestamp(2000, 1, 1, 0, 0, 0, 0, pytz.UTC, 1)
```

Reply:
make your test
``2000, 1, 2, 3, 4, 5, 6``